tests/integration/crunchers/test_cruncher_quantile_rolling_windows.py

```diff
@@ -214,6 +219,41 @@ def test_numerical_relationship(self, use_ratio):
             expected = interpolate_fn(xs_to_interp)
         assert all(crunched["value"].values == expected)
 
+    def test_limit_of_similar_xs(self, test_db):
+        # Check that the function returns the correct behaviour in the limit of similar
+        # lead values. We construct a db for exactly identical CH4 and for near-
+        # identical CH4.
+        range_db = pd.DataFrame(
+            [[_ma, str(val), "World", _ech4, _gtc, val] for val in range(10)],
+            columns=_msrvu + [2010],
+        )
+        range_db = IamDataFrame(range_db)
+        range_db = self._adjust_time_style_to_match(range_db, test_db)
+        same_db = range_db.copy()
+        same_db["variable"] = _eco2
+        same_db["value"] = 1
+        same_db.append(range_db, inplace=True)
+        nearly_same_db = same_db.copy()
+        # We change the value of one point on the nearly_same and remove it on the same
+        nearly_same_db.data["value"].iloc[1] = 0
+        same_db.filter(scenario="0", keep=False, inplace=True)
+        same_db = IamDataFrame(same_db.data)
+        nearly_same_db = IamDataFrame(nearly_same_db.data)
```

Comment:
why do you need to reinitialise the `IamDataFrame`'s here?

Reply:
As I said above, I just always do that because I've run into so many bugs due to having unexpected values in columns.